scenarios/validation_receipts/src/main.rs

```diff
@@ -0,0 +1,134 @@
+use holochain_types::prelude::*;
+use std::time::{Duration, Instant};
+use trycp_wind_tunnel_runner::embed_conductor_config;
+use trycp_wind_tunnel_runner::prelude::*;
+
+embed_conductor_config!();
+
+#[derive(Debug, Default)]
+pub struct ScenarioValues {}
+
+impl UserValuesConstraint for ScenarioValues {}
+
+fn agent_setup(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+) -> HookResult {
+    connect_trycp_client(ctx)?;
+    reset_trycp_remote(ctx)?;
+
+    let client = ctx.get().trycp_client();
+    let agent_name = ctx.agent_name().to_string();
+
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            client
+                .configure_player(agent_name.clone(), conductor_config().to_string(), None)
+                .await?;
+
+            client.startup(agent_name.clone(), None).await?;
+
+            Ok(())
+        })?;
+
+    install_app(ctx, scenario_happ_path!("crud"), &"crud".to_string())?;
+    try_wait_for_min_peers(ctx, Duration::from_secs(120))?;
+
+    Ok(())
+}
+
+fn agent_behaviour(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+) -> HookResult {
+    let reporter = ctx.runner_context().reporter();
+
+    let action_hash: ActionHash = call_zome(
+        ctx,
+        "crud",
+        "create_sample_entry",
+        "this is a test entry value",
+        Some(Duration::from_secs(80)),
+    )?;
+
+    let response: Option<Record> = call_zome(
+        ctx,
+        "crud",
+        "get_sample_entry",
+        action_hash.clone(),
+        Some(Duration::from_secs(80)),
+    )?;
+
+    assert!(response.is_some(), "Expected record to be found");
+
+    let start = Instant::now();
+
+    'outer: loop {
```

Comment:
Busy loop? I would say that it's acceptable to sleep for 10-25ms or something around there. It does affect the results slightly but I'd rather we correct for that in the future by recording timestamps on the validation receipts so we can get an accurate receive time without needing to query fast.

Reply:
removed